### PR TITLE
ViewModel that DataBinding apply

### PR DESCRIPTION
**Pull request by lcyer**
Just See  My Work...and Please. My Work Review

This Work Need New Layout That RecyclerView Item...

i try AAC, Layout Renewal
 
p.s layout so bad / very bad

### CODE REVIEW

File: app/src/main/java/com/cmdevs/projectunknown/util/FragmentUtil.kt [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+import android.support.annotation.IdRes
+import android.support.annotation.LayoutRes
+import android.support.v4.app.Fragment
+import android.support.v4.app.FragmentManager
+
+
+fun FragmentManager.fragReplace(@IdRes id: Int, fragment: Fragment, tag: String? = null){
+    this.beginTransaction().replace(id, fragment, tag).commit()
+}
```
Comment (jukqaz) [inline app/src/main/java/com/cmdevs/projectunknown/util/FragmentUtil.kt]: @lcyer 
Very nice!
And i suggest other expression. 
`AppCompatActivity.replaceFragment(@IdRes id:Int, fragment:Fragment, tag:String? = null) {
   suuportFragmentManager.beginTransaction().replace(id, fragment, tag).commit()
}`

File: app/src/main/java/com/cmdevs/projectunknown/view/friend/FriendFragment.kt [hunk 2]
```diff
@@ -2,13 +2,32 @@ package com.cmdevs.projectunknown.view.friend
...
+    private val friendItemViewModel: FriendItemViewModel by lazy {
+        FriendItemViewModel(FriendListRepository)
+    }
+
+    private val friendListAdapter: FriendListAdapter by lazy {
+        FriendListAdapter(friendItemViewModel)
+    }
+
+    companion object {
+        fun getInstance() = FriendFragment()
```
Comment (jukqaz) [inline app/src/main/java/com/cmdevs/projectunknown/view/friend/FriendFragment.kt]: @lcyer 
I think use `newInstance()` better than `getInstance`.

File: app/src/main/java/com/cmdevs/projectunknown/view/friend/FriendItemViewModel.kt [hunk 3]
```diff
@@ -0,0 +1,48 @@
+package com.cmdevs.projectunknown.view.friend
+
+import android.databinding.BaseObservable
+import android.databinding.Bindable
+import android.databinding.Observable
+import android.databinding.ObservableField
+import com.cmdevs.projectunknown.data.FriendListData
+import com.cmdevs.projectunknown.data.friend.FriendListRepository
+
+class FriendItemViewModel(val friendListRepository: FriendListRepository) : BaseObservable() {
```
Comment (jukqaz) [inline app/src/main/java/com/cmdevs/projectunknown/view/friend/FriendItemViewModel.kt]: @lcyer 
I suggest that remove `friendListRepository` If you not use it.

File: app/src/main/java/com/cmdevs/projectunknown/view/friend/FriendViewModel.kt [hunk 4]
```diff
@@ -1,22 +1,18 @@
-package com.cmdevs.projectunknown.view.friend.viewmodel
+package com.cmdevs.projectunknown.view.friend
 
 import com.cmdevs.projectunknown.data.friend.FriendListRepository
 import com.cmdevs.projectunknown.view.friend.adapter.model.FriendRecyclerModel
 
 class FriendViewModel(
     private val friendListRepository: FriendListRepository,
-    private val firendRecyclerModel: FriendRecyclerModel) : ViewModel{
+    private val firendRecyclerModel: FriendRecyclerModel) {
```
Comment (jukqaz) [inline app/src/main/java/com/cmdevs/projectunknown/view/friend/FriendViewModel.kt]: @lcyer 
I think that fix typo `firend`.

File: app/src/main/java/com/cmdevs/projectunknown/view/friend/adapter/FriendListAdapter.kt [hunk 5]
```diff
@@ -29,4 +42,5 @@ class FriendListAdapter(private val context: Context): RecyclerView.Adapter<Recy
         notifyDataSetChange()
```
Comment (jukqaz) [inline app/src/main/java/com/cmdevs/projectunknown/view/friend/adapter/FriendListAdapter.kt]: @lcyer 
I think that code have to remove.

File: app/src/main/java/com/cmdevs/projectunknown/util/FragmentUtil.kt (hunk 1 above)
Comment (lcyer) [reply app/src/main/java/com/cmdevs/projectunknown/util/FragmentUtil.kt]: Thanks 
That`s nice better than my Util code.
got it!

File: app/src/main/java/com/cmdevs/projectunknown/view/friend/FriendFragment.kt (hunk 2 above)
Comment (lcyer) [reply app/src/main/java/com/cmdevs/projectunknown/view/friend/FriendFragment.kt]: Thanks.
i try rename that getInstance() code name

File: app/src/main/java/com/cmdevs/projectunknown/view/friend/FriendItemViewModel.kt (hunk 3 above)
Comment (lcyer) [reply app/src/main/java/com/cmdevs/projectunknown/view/friend/FriendItemViewModel.kt]: thank`s for comment

i think that code later need in FriendItemViewModel

File: app/src/main/java/com/cmdevs/projectunknown/view/friend/FriendViewModel.kt (hunk 4 above)
Comment (lcyer) [reply app/src/main/java/com/cmdevs/projectunknown/view/friend/FriendViewModel.kt]: got it.
i try fix code name

**Pull request merged**